code/parachain/frame/assets-transactor-router/src/lib.rs

```diff
@@ -0,0 +1,551 @@
+			#[pallet::compact] amount: T::Balance,
+			governance_origin: <T::Lookup as StaticLookup>::Source,
+			dest: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			let id = <Self as CreateAsset>::create_local_asset(
+				protocol_id,
+				nonce,
+				name,
+				symbol,
+				decimals,
+				ratio,
+			)?;
+			let governance_origin = T::Lookup::lookup(governance_origin)?;
+			T::GovernanceRegistry::set(id, SignedRawOrigin::Signed(governance_origin));
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::mint_into(id, &dest, amount)?;
+			Ok(())
+		}
+
+		/// Mints `amount` of `asset_id` into the `dest` account.
+		#[pallet::weight(T::WeightInfo::mint_into())]
+		pub fn mint_into(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			dest: <T::Lookup as StaticLookup>::Source,
+			#[pallet::compact] amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::mint_into(asset_id, &dest, amount)?;
+			Ok(())
+		}
+
+		/// Burns `amount` of `asset_id` into the `dest` account.
+		#[pallet::weight(T::WeightInfo::burn_from())]
+		pub fn burn_from(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			dest: <T::Lookup as StaticLookup>::Source,
+			#[pallet::compact] amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::burn_from(asset_id, &dest, amount)?;
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns `Ok(())` if origin is root or asset is signed by root or by origin
+		pub(crate) fn ensure_admin_or_governance(
+			origin: OriginFor<T>,
+			asset_id: &T::AssetId,
+		) -> Result<(), DispatchError> {
+			// TODO: that must be ensure_asset_origin(origin, asset_id))
+			if T::AdminOrigin::ensure_origin(origin.clone()).is_ok() {
+				return Ok(())
+			}
+
+			match origin.into() {
+				Ok(frame_system::RawOrigin::Signed(account)) => {
+					match T::GovernanceRegistry::get(asset_id) {
+						Ok(SignedRawOrigin::Root) => Ok(()), /* ISSUE: it says if */
+						// (call_origin.is_signed &&
+						// asst_owner.is_root) then allow
+						// mint/burn -> anybody can mint and
+						// burn PICA?
+						// TODO: https://app.clickup.com/t/37h4edu
+						Ok(SignedRawOrigin::Signed(acc)) if acc == account => Ok(()),
+						_ => Err(DispatchError::BadOrigin),
+					}
+				},
+				Ok(frame_system::RawOrigin::Root) => Ok(()),
+				_ => Err(DispatchError::BadOrigin), /* ISSUE: likely will not support collective
+													* origin which is reasonable to have for
+													* governance
+													https://app.clickup.com/t/37h4edu
+													*/
+			}
+		}
+	}
+
+	impl<T: Config> CreateAsset for Pallet<T> {
+		type LocalAssetId = T::AssetId;
+		type ForeignAssetId = T::AssetLocation;
```

Comment:
Asset creation trait enabling other protocols to create assets in the preferred way